test/test_pairs_classifiers.py

```diff
@@ -66,14 +66,24 @@ def test_predict_monotonous(estimator, build_dataset,
                          ids=ids_pairs_learners)
 def test_raise_not_fitted_error_if_not_fitted(estimator, build_dataset,
                                               with_preprocessor):
-  """Test that a NotFittedError is raised if someone tries to predict and
-  the metric learner has not been fitted."""
+  """Test that a NotFittedError is raised if someone tries to use
+  score_pairs, decision_function, get_metric, transform or
+  get_mahalanobis_matrix on input data and the metric learner
+  has not been fitted."""
   input_data, labels, preprocessor, _ = build_dataset(with_preprocessor)
   estimator = clone(estimator)
   estimator.set_params(preprocessor=preprocessor)
   set_random_state(estimator)
+  with pytest.raises(NotFittedError):
+    estimator.score_pairs(input_data)
   with pytest.raises(NotFittedError):
     estimator.decision_function(input_data)
+  with pytest.raises(NotFittedError):
+    estimator.get_metric()
+  with pytest.raises(NotFittedError):
+    estimator.transform(input_data)
+  with pytest.raises(NotFittedError):
+    estimator.get_mahalanobis_matrix()
```

Comment:
actually:
- you should also add `calibrate_threshold` here
- shouldn't we also include `predict` at the end (after calling `set_threshold` so that the threshold is set)?
- also, isn't a single `with` enough?

Reply:
* I did not add `calibrate_threshold` because it calls `decision_function`, but I added it now.
* Ok, done.
* I don't agree, since with all the different `with`'s we test whether each test returns the right error, while one `with` would check whether an error is raised and if it's the right one. If any of the tested methods passes without an error, several `with`'s detect that, while only one doesn't.

Reply:
Indeed, my mistake :-)